traefik/tests/deployment-config_test.yaml

```diff
@@ -84,6 +84,14 @@ tests:
       - equal:
           path: spec.revisionHistoryLimit
           value: 1
+  - it: should have revisionHistoryLimit set to 0 when specified as 0
```

Comment:
Please add a test showing `revisionHistoryLimit` is not set when `.Values.deployment.revisionHistoryLimit` is not set (which is the purpose of this PR).